server/db/interface.go

```diff
@@ -14,8 +40,14 @@ import (
 // DEXArchivist will be composed of several different interfaces. Starting with
 // OrderArchiver.
 type DEXArchivist interface {
+	// LastErr should returns any fatal or unexpected error encountered by the
+	// archivist backend. This may be used to check if the database had an
+	// unrecoverable error (disconnect, etc.).
+	LastErr() error
```

Comment:
Since there are multiple packages sharing the archivist, we need a solution for communicating irrecoverable database failures to all consumers (esp. Market or higher level), such as:
- disconnected database
- a unexpected failure to store or retrieve data that likely disrupts future operations

This is just a first shot at that, and it needs more thought.